pkg/retry/retry.go

```diff
@@ -98,3 +93,11 @@ func isRetryable(err error) bool {
 
 	return false
 }
+
+func shouldRestart(e error) bool {
```

Comment:
This introduces `isRetryable(error)` and `shouldRestart(error)` with exactly the same signature and pretty much the same name, and no documentation as to intent. Having the second one an `isErrnoRetryable(syscall.Errno)` or something like that would be clearer.